### ISSUE TITLE
Exit code on failure

### ISSUE THREAD

**Issue by LolHens** [itzg/mc-server-runner#7]
Hi! On my docker swarm when minecraft exits with an error it seems like mc-server-runner just returns exit code 0.
Docker then displays my container not as failed but as completed.
In the log I see the following:
```
[13:05:23] [main/FATAL]: Failed to start the minecraft server
java.lang.RuntimeException: ...
2020-08-04T13:05:23.996+0200	INFO	mc-server-runner	Done
```
![image](https://user-images.githubusercontent.com/1524059/89293842-e0805c80-d65e-11ea-8d44-1013beec01b4.png)

Could this be because of mc-server-runner?

**Comment by itzg** [comment]
Ah yes, I think I see where it is discarding the sun-process exit code.

**Comment by itzg** [comment]
I'm about to push an update to `itzg/minecraft-server` that includes the new mc-server-runner (1.5.0); however, I'm starting to wonder if we're stuck with exit code 0 since the java process doesn't seem to report otherwise in this test:

```
bash-4.4# java -jar minecraft_server.1.16.1.jar --not-valid
[18:26:46] [main/FATAL]: Failed to start the minecraft server
joptsimple.UnrecognizedOptionException: not-valid is not a recognized option
        at joptsimple.OptionException.unrecognizedOption(OptionException.java:108) ~[minecraft_server.1.16.1.jar:?]
        at joptsimple.OptionParser.handleLongOptionToken(OptionParser.java:510) ~[minecraft_server.1.16.1.jar:?]
        at joptsimple.OptionParserState$2.handleArgument(OptionParserState.java:56) ~[minecraft_server.1.16.1.jar:?]
        at joptsimple.OptionParser.parse(OptionParser.java:396) ~[minecraft_server.1.16.1.jar:?]
        at net.minecraft.server.Main.main(SourceFile:80) [minecraft_server.1.16.1.jar:?]
bash-4.4# echo $?
0
```

**Comment by itzg** [comment]
Latest `itzg/minecraft-server` image includes this change, so let's see if your particular error scenario happens to pop out of java with a non-zero exit code.

**Comment by LolHens** [comment]
The new version fixes the problem for me! The container now exits with status "failed" like it should.
Here is an example:
```
2020-08-09T19:54:16.843382454Z [init] Running as uid=1000 gid=1000 with /data as 'drwxrwxrwx    9 1000     1000            23 Aug  9 13:59 /data'
2020-08-09T19:54:16.848618223Z [init] Resolved version given 1.16.1 into 1.16.1
2020-08-09T19:54:16.848845685Z [init] Resolving type given FABRIC
2020-08-09T19:54:16.851670161Z [init] Checking Fabric version information.
2020-08-09T19:54:16.863645825Z [init] server.properties already created, skipping
2020-08-09T19:54:16.868621074Z [init] log4j2.xml already created, skipping
2020-08-09T19:54:16.868650409Z [init] Checking for JSON files.
2020-08-09T19:54:16.889645574Z [init] Setting initial memory to 4G and max to 4G
2020-08-09T19:54:16.890723319Z [init] Starting the Minecraft server...
2020-08-09T19:54:20.063485645Z [21:54:20] [main/INFO]: Loading for game Minecraft 1.16.1
2020-08-09T19:54:21.492256008Z [21:54:21] [main/FATAL]: A critical error occurred
2020-08-09T19:54:21.492364999Z 	... 12 more
2020-08-09T19:54:21.551382344Z 2020-08-09T21:54:21.548+0200	WARN	mc-server-runner	sub-process failed	{"exitCode": 1}
2020-08-09T19:54:21.551410223Z 2020-08-09T21:54:21.549+0200	INFO	mc-server-runner	Done
```
